pkg/apis/agent/v1/klusterletaddonconfig_types.go

```diff
@@ -64,31 +67,47 @@ type KlusterletAddonConfigApplicationManagerSpec struct {
 
 	// +optional
 	ArgoCDCluster bool `json:"argocdCluster"`
+
+	// +optional
+	ProxyConfig ProxyConfig `json:"proxyConfig"`
```

Comment:
I would not add the config here. I might add a ProxyConfig in Spec with a list of addon names to use addons.

Reply:
done